pkg/cli/login/loginoptions.go

```diff
@@ -136,6 +137,10 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 	// try to TCP connect to the server to make sure it's reachable, and discover
 	// about the need of certificates or insecure TLS
 	if err := dialToServer(*clientConfig); err != nil {
+		if strings.Contains(err.Error(), "certificate is not trusted") {
+			err = x509.CertificateInvalidError{Reason: x509.Expired, Detail: err.Error()}
```

Comment:
Is not this actually going to hide the error? What's the difference between the current behavior vs. the behavior after merging the changes? Is the idea here to provide a better formatted error string? Or, will this change lead to a different execution of the code?

Reply:
Creating the `x509.CertificateInvalidError` causes the `switch case` to catch this like any other invalid certificate issue. The idea was to use the best descriptive `error` type, and unfortunately, `x509.CertificateInvalidError.Error()` method only outputs the `Detail` string when a certain `Reason` is set. In this case, `x509.Expired` was the closest reason I could find.

I haven't found anything in kubectl that handles this yet, but oc and kubectl have very different authentication mechanisms. I agree that reusing `x509.CertificateInvalidError` in such a way is not ideal. I was trying to keep the number of changes as small as possible, but I may have to create a custom error for this condition to better handle it and reduce the user confusion of getting both an "expired" and  "not trusted" message in the same error.

Reply:
I tested this on a Mac, and it does resolve the issue for logins, I'm not 100% sure about other commands though. It should be possible to replicate this fix globally if it is an issue outside of login.

Reply:
Based on https://go-review.googlesource.com/c/go/+/418835 the Go release cycle entered the code freeze phase (Nov-Jan). So I don't expect the fix getting reviews anytime soon. Though, it's still worth reaching the upstream Go community to get their attention so they are aware of the code change request. It seems noone is aware of the PR. @deejross worth checking the thread in https://groups.google.com/g/golang-dev/c/K7oGURi0wTM/m/HTyfcmY0BwAJ. Comment from Ian in https://groups.google.com/g/golang-dev/c/K7oGURi0wTM/m/HTyfcmY0BwAJ:
```
In general it should not be necessary to ask for an additional Googler
review. If your CL shows up on https://go.dev/s/needs-review, then
some Googler will get to it soon.

If your CL does not show up on that list, then most likely it has an
unresolved comment, or the trybots have not been run.

Please do ping if your CL is not on that list and you don't understand
why, or if it has been on the list for several days. But in general
pinging should not be required.
```

Reply:
When it comes to resolving this issue. @deejross if you are confident this change fixes the login issue without any negative side effects, let's proceed. @kasturinarra would you please perform pre-merge testing on all architectures for the `oc login`? Just to be sure we are not breaking anything.

Reply:
I think, we can put a TODO explaining why we did this and when we can remove this hacky solution with the links.

Reply:
@ingvagabund thanks, i will ask @zhouying7780 to take a look at it, thanks !!